src/reviewRenderer/renderers/interactions/TextEntryInteraction.js

```diff
@@ -17,10 +17,148 @@
  *
  */
 
+import _ from 'lodash';
+import template from 'taoQtiItem/reviewRenderer/tpl/interactions/textEntryInteraction';
+import containerHelper from 'taoQtiItem/qtiCommonRenderer/helpers/container';
+import pciResponse from 'taoQtiItem/qtiCommonRenderer/helpers/PciResponse';
+import locale from 'util/locale';
+
 /**
- * @author Ansul Sharma <ansultaotesting.com>
+ * Init rendering, called after template injected into the DOM
+ * All options are listed in the QTI v2.1 information model:
+ * http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element10333
+ *
+ * @param {object} interaction
  */
-import template from 'taoQtiItem/reviewRenderer/tpl/interactions/textEntryInteraction';
-import textEntryInteraction from 'taoQtiItem/qtiCommonRenderer/renderers/interactions/TextEntryInteraction';
+const render = (interaction) => {
+    const attributes = interaction.getAttributes();
+
+    //setting up the width of the input field
+    if (attributes.expectedLength) {
+        const $input = interaction.getContainer();
+
+        //adding 2 chars to include reasonable padding size
+        const expectedLength = parseInt(attributes.expectedLength) + 2;
+        $input.css({
+            width: expectedLength + 'ch',
+            minWidth: expectedLength + 'ch'
+        });
+    }
+};
+
+const resetResponse = interaction => {
+    interaction.getContainer().text('');
+};
+
+/**
+ * Set the response to the rendered interaction.
+ *
+ * The response format follows the IMS PCI recommendation :
+ * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+ *
+ * Available base types are defined in the QTI v2.1 information model:
+ * http://www.imsglobal.org/question/qtiv2p1/imsqti_infov2p1.html#element10333
+ *
+ * Special value: the empty object value {} resets the interaction responses
+ *
+ * @param {object} interaction
+ * @param {object} response
+ */
+const setResponse = (interaction, response) => {
+    let responseValue;
+
+    try {
+        responseValue = pciResponse.unserialize(response, interaction);
+    } catch (e) {}
```

Comment:
I guess we should add console.error at least

Reply:
Just to keep align with the common renderer: https://github.com/oat-sa/tao-item-runner-qti-fe/blob/16d475c8ef51f8b03b3a3047dfcfab4adce0c831/src/qtiCommonRenderer/renderers/interactions/TextEntryInteraction.js#L202
But can be added if necessary